packages/changed-elements-react/src/widgets/comparisonJobWidget/hooks/useNamedVersionLoader.tsx

```diff
@@ -175,176 +157,65 @@ const sortAndSetIndexOfNamedVersions = (namedVersions: NamedVersion[], currentNa
     onError();
     return;
   }
-  const reversedNamedVersions = namedVersionsOlderThanCurrentVersion.reverse()
+  const reversedNamedVersions = namedVersionsOlderThanCurrentVersion.reverse();
   if (reversedNamedVersions[0].changesetIndex === currentNamedVersion.changesetIndex) {
     reversedNamedVersions.shift(); //remove current named version
   }
   const changeSetMap = new Map<number, Changeset>();
   changesets.forEach((changeset: Changeset) => {
     changeSetMap.set(changeset.index, changeset);
-  })
+  });
   // we must offset the named versions , because that changeset is "already applied" to the named version, see this:
   // https://developer.bentley.com/tutorials/changed-elements-api/#221-using-the-api-to-get-changed-elements
   // this assuming latest is current
   const offSetNameVersions = reversedNamedVersions.map((version) => {
     version.changesetIndex = version.changesetIndex + 1;
-    version.changesetId = changeSetMap.get(version.changesetIndex)?.id ?? version.changesetId
+    version.changesetId = changeSetMap.get(version.changesetIndex)?.id ?? version.changesetId;
     return version;
-  })
+  });
   return offSetNameVersions;
 };
 
 type ProcessChangesetsArgs = {
   iTwinId: string;
   iModelId: string;
-  namedVersionLoaderState: NamedVersionLoaderState;
+  namedVersionLoaderState: NamedVersionLoaderResult;
   iModelConnection: IModelConnection;
   comparisonJobClient: IComparisonJobClient;
   setResult: (result: NamedVersionLoaderResult) => void;
-  isDisposed: () => boolean;
-};
-
-type Entry = {
-  version: NamedVersion;
-  state: VersionProcessedState;
-  jobStatus: JobStatus;
-  jobProgress: JobProgress;
+  getPendingJobs: () => JobAndNamedVersions[],
 };
 
 const processChangesetsAndUpdateResultState = async (args: ProcessChangesetsArgs) => {
-  const currentVersionId = args.namedVersionLoaderState.result.namedVersions.currentVersion.version.changesetId ??
+  const pendingJobsMap = new Map<JobId, JobAndNamedVersions>();
+  args.getPendingJobs().forEach((job) => {
+    pendingJobsMap.set(`${job.targetNamedVersion.changesetId}-${job.currentNamedVersion.changesetId}`, job);
+  });
+  const currentVersionId = args.namedVersionLoaderState.namedVersions.currentVersion?.version.changesetId ??
     args.iModelConnection?.changeset.id;
-  const newEntries = await Promise.all(args.namedVersionLoaderState.result.namedVersions.entries.map(async (entry) => {
+  const newEntries = await Promise.all(args.namedVersionLoaderState.namedVersions.entries.map(async (entry) => {
     const jobStatusAndJobProgress: JobStatusAndJobProgress = await getJobStatusAndJobProgress(args.comparisonJobClient, entry, args.iTwinId, args.iModelId, currentVersionId);
+    if (pendingJobsMap.has(`${entry.version.changesetId}-${currentVersionId}`)) {
+      const jobStatus:JobStatus="Processing"
```

Comment:
Spacing and `;`
```suggestion
      const jobStatus: JobStatus = "Processing";
```
Maybe run the lint script in the repo?